src/main/java/seedu/address/storage/JsonAdaptedTransaction.java

```diff
@@ -1,26 +1,28 @@
 package seedu.address.storage;
 
+import java.time.LocalDate;
+
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import seedu.address.model.person.Transaction;
 
 /**
- * Json for {@code Transaction}
+ * Json for {@code Transaction}.
  */
 public class JsonAdaptedTransaction {
     private final String description;
     private final int amount;
     private final String otherParty;
-    private final String date;
+    private final LocalDate date;
 
     /**
      * Constructs a {@code JsonAdaptedTransaction} with the given {@code description},
      * {@code amount}, {@code otherParty} and {@code date}.
      */
     @JsonCreator
     public JsonAdaptedTransaction(@JsonProperty("description") String description, @JsonProperty("amount") int amount,
-                                  @JsonProperty("otherParty") String otherParty, @JsonProperty("date") String date) {
+                          @JsonProperty("otherParty") String otherParty, @JsonProperty("date") LocalDate date) {
```

Comment:
Did you test if the storage saves and reads the dates correctly?

Reply:
Have tested, both saving and loading are compatible with the changes